devops/azure-pipelines.yml

```diff
@@ -79,3 +79,28 @@ stages:
 
   displayName: Dockerize --> Stage
 
+- stage: deploy
```

Comment:
this stage needs to not run for PR builds. you can see yourself at the bottom of the conversation tab that the deploy stage has already run on this branch. this is a big oversight that could cause more serious problems in more serious environments. the danger here comes from the catch-22 of PR builds - if you change the PR build logic in a PR, it's going to run before it can be reviewed, at least with a simple setup. so you have to be extra careful about it.

Reply:
Resolved - pipelines now only deploy on commit, and only tests and builds on PR